packages/grafana-schema/src/schema/dashboard/dashboard.gen.ts

```diff
@@ -0,0 +1,216 @@
+// This file is autogenerated. DO NOT EDIT.
+//
+// Derived from the Thema lineage at pkg/coremodel/dashboard/lineage.cue
+
+export interface AnnotationQuery {
+  builtIn: number;
+  datasource: {};
+  enable: boolean;
+  hide?: boolean;
+  iconColor?: string;
+  name?: string;
+  rawQuery?: string;
+  showIn: number;
+  target?: {};
+  type: string;
+}
+
+export const defaultAnnotationQuery: Partial<AnnotationQuery> = {
+  builtIn: 0,
+  enable: true,
+  hide: false,
+  showIn: 0,
+  type: 'dashboard',
+};
+
+export interface VariableModel {
+  label?: string;
+  name: string;
+  type: VariableType;
+}
+
+export interface DashboardLink {
+  asDropdown: boolean;
+  icon?: string;
+  includeVars: boolean;
+  keepTime: boolean;
+  tags: string[];
+  targetBlank: boolean;
+  title: string;
+  tooltip?: string;
+  type: DashboardLinkType;
+  url?: string;
+}
+
+export const defaultDashboardLink: Partial<DashboardLink> = {
+  asDropdown: false,
+  includeVars: false,
+  keepTime: false,
+  tags: [],
+  targetBlank: false,
+};
+
+export type DashboardLinkType = 'link' | 'dashboards';
+
+export type VariableType = 'query' | 'adhoc' | 'constant' | 'datasource' | 'interval' | 'textbox' | 'custom' | 'system';
+
+export enum FieldColorModeId {
+  ContinuousGrYlRd = 'continuous-GrYlRd',
+  Fixed = 'fixed',
+  PaletteClassic = 'palette-classic',
+  PaletteSaturated = 'palette-saturated',
+  Thresholds = 'thresholds',
+}
```

Comment:
just an FYI these is just some of the names, there are many more that are not given a specific enum value

Reply:
does that mean the actual type of a field is effectively `FieldColorModeID | string`?